src/main/java/me/earth/earthhack/impl/modules/player/ncptweaks/ListenerPosLook.java

```diff
@@ -0,0 +1,33 @@
+package me.earth.earthhack.impl.modules.player.ncptweaks;
+
+import me.earth.earthhack.impl.core.mixins.entity.living.IEntityFireworkRocket;
+import me.earth.earthhack.impl.event.events.network.PacketEvent;
+import me.earth.earthhack.impl.event.listeners.ModuleListener;
+import net.minecraft.entity.item.EntityFireworkRocket;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemElytra;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.play.server.SPacketPlayerPosLook;
+
+final class ListenerPosLook extends ModuleListener<NCPTweaks, PacketEvent.Receive<SPacketPlayerPosLook>> {
+    public ListenerPosLook(NCPTweaks module) {
+        super(module, PacketEvent.Receive.class);
+    }
+
+    @Override
+    public void invoke(PacketEvent.Receive<SPacketPlayerPosLook> event) {
+        if (!module.elytraFix.getValue()) {
+            return;
+        }
+
+        ItemStack stack = mc.player.getItemStackFromSlot(EntityEquipmentSlot.CHEST);
+
+        if (stack.getItem() == Items.ELYTRA && ItemElytra.isUsable(stack) && mc.player.isElytraFlying()) {
+            for (EntityFireworkRocket e : mc.world.getEntities(EntityFireworkRocket.class, e -> ((IEntityFireworkRocket)e).getBoostedEntity() == mc.player)) {
```

Comment:
PacketEvent.Receive gets called on the NetworkThread, not on the main thread, making all of this unsafe. Can you wrap all of this with `mc.addScheduledTask`, inside a null check for mc.player and the world?